app/javascript/dashboard/components/widgets/conversation/ConversationBulkActions.vue

```diff
@@ -0,0 +1,280 @@
+                })
+              }}
+              <strong>
+                {{ selectedAgent.name }}
+              </strong>
+            </p>
+            <div class="agent-confirmation-actions">
+              <woot-button
+                color-scheme="primary"
+                variant="smooth"
+                @click="goBack"
+              >
+                {{ $t('BULK_ACTION.GO_BACK_LABEL') }}
+              </woot-button>
+              <woot-button
+                color-scheme="primary"
+                variant="flat"
+                :is-loading="uiFlags.isUpdating"
+                @click="submit"
+              >
+                {{ $t('BULK_ACTION.ASSIGN_LABEL') }}
+              </woot-button>
+            </div>
+          </div>
+        </div>
+      </div>
+    </transition>
+  </div>
+</template>
+
+<script>
+import Thumbnail from 'dashboard/components/widgets/Thumbnail.vue';
+import { mapGetters } from 'vuex';
+
+export default {
+  components: {
+    Thumbnail,
+  },
+  props: {
+    conversations: {
+      type: Array,
+      default: () => [],
+    },
+    allConversationsSelected: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  data() {
+    return {
+      showAgentsList: false,
+      query: '',
+      selectedAgent: null,
+    };
+  },
+  computed: {
+    ...mapGetters({
+      agents: 'agents/getAgents',
+      uiFlags: 'bulkActions/getUIFlags',
+    }),
+    filteredAgents() {
+      return this.agents.filter(agent =>
+        agent.name.toLowerCase().includes(this.query.toLowerCase())
+      );
+    },
+  },
+  mounted() {
+    this.$refs.selectAllCheck.indeterminate = true;
+  },
+  methods: {
+    assignAgent(agent) {
+      this.selectedAgent = agent;
+    },
+    selectAll(e) {
+      this.$emit('selectAllConversations', e.target.checked);
+    },
+    goBack() {
+      this.selectedAgent = null;
+    },
+    submit() {
+      this.$emit('assignAgent', this.selectedAgent);
+    },
+    resolveConversations() {
+      this.$emit('resolveConversations');
+    },
+  },
+};
+</script>
+
+<style scoped lang="scss">
+.flex-between {
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+}
+.bulk-action__container {
+  padding: var(--space-normal) var(--space-one);
+  background-color: var(--s-50);
+  position: relative;
+  box-shadow: 6px 3px 22px 9px rgb(181 181 181 / 50%);
+}
+.search-container {
+  padding: 0 var(--space-one);
+  position: sticky;
+  top: 0;
+  z-index: var(--z-index-twenty);
+  background-color: var(--white);
+}
+
+.bulk-action__panel {
+  cursor: pointer;
+  span {
+    font-size: var(--font-size-mini);
+    margin-left: var(--space-smaller);
+  }
+  input[type='checkbox'] {
+    margin: var(--space-zero);
+    cursor: pointer;
+  }
+}
+
+.mr-small {
+  margin-right: var(--space-smaller);
+}
+
+.bulk-action__agents {
+  position: absolute;
+  bottom: 40px;
+  right: var(--space-small);
+  width: 100%;
+  box-shadow: 0 0.8rem 1.6rem rgb(50 50 93 / 8%),
```

Comment:
Can we use shadow as variable from here `app/javascript/shared/assets/stylesheets/shadows.scss`?